plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/index/HudiBaseIndexSupport.java

```diff
@@ -58,4 +61,14 @@ protected Map<String, HoodieIndexDefinition> getAllIndexDefinitions()
 
         return lazyMetaClient.get().getIndexMetadata().get().getIndexDefinitions();
     }
+
+    protected String getDatabaseName()
+    {
+        return lazyMetaClient.get().getTableConfig().getDatabaseName();
```

Comment:
This can be different from the catalog table's database name.  We should get the database name from the table handle.

Reply:
Fixed.